From 8be515beb891e36449b4ab2e51b0c4b8a7c5db8c Mon Sep 17 00:00:00 2001
From: Martin Kletzander <mkletzan@redhat.com>
Date: Mon, 29 Jul 2013 08:33:30 +0200
Subject: [RHEL-6.5 python-virtinst PATCH 2/5] Fix progress bars in virt-clone
 once more

https://bugzilla.redhat.com/show_bug.cgi?id=946972

Upstream commit 665375db was meant to fix the progress bar problems,
but as I've found out just made one race (which still persisted) a
little less probable.  The real problem was in the upper code which
meant to finish the progress bar properly, however this was not done
thanks to 'finally' statement being not executed when the function
returned in the 'try' part.  This patch makes the upper caller wait
for the updating thread to finish and then properly end()s the
progress bar's output.

(cherry picked from commit 5208c5f321a8e81091027cfc835520fc3a4f7bcf)

Conflicts:
	virtinst/Storage.py -- try->except->finally is still
                            try->try->except in old python-virtinst

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
---
 virtinst/Storage.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/virtinst/Storage.py b/virtinst/Storage.py
index a1c2eec..47ecc3c 100644
--- a/virtinst/Storage.py
+++ b/virtinst/Storage.py
@@ -1212,6 +1212,8 @@ class StorageVolume(StorageObject):
                 else:
                     vol = self.pool.createXML(xml, 0)

+                self._install_finished = True
+                t.join()
                 if meter:
                     meter.end(self.capacity)
                 logging.debug("Storage volume '%s' install complete." %
-- 
1.8.3.2

